internal/kafka/internal/services/kafkatlscertmgmt/memory_storage.go

```diff
@@ -40,7 +41,11 @@ func newInMemoryStorage() *inMemoryStorage {
 func (storage *inMemoryStorage) Lock(ctx context.Context, key string) error {
```

Comment:
my suggestion will be to re-use what we've here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/9d14b767dc88eebaee00867bf57f44f8ad4873e0/internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go#L64-L74 

and remove any locking related details from the stored item.